whisper/src/bin/find-corrupt-whisper-files.rs

```diff
@@ -0,0 +1,92 @@
+#[macro_use]
+extern crate structopt;
+extern crate failure;
+extern crate walkdir;
+extern crate whisper;
+
+use failure::Error;
+use std::fs::remove_file;
+use std::path::{PathBuf, Path};
+use structopt::StructOpt;
+use walkdir::WalkDir;
+
+
+#[derive(Debug, StructOpt)]
+#[structopt(
+    name = "find-corrupt-whisper-files",
+    about = "Find and (optionally) delete corrupt Whisper data files."
+)]
+struct Args {
+    /// Delete reported files.
+    #[structopt(long = "delete-corrupt", help = "Delete reported files")]
+    delete_corrupt: bool,
+
+    /// Display progress info.
+    #[structopt(long = "verbose", help = "Display progress info")]
+    verbose: bool,
+
+    /// Directory containing Whisper files.
+    #[structopt(
+        name = "WHISPER_DIR",
+        help = "Directory containing Whisper files",
+        parse(from_os_str),
+        raw(required = "true", min_values = "1")
+    )]
+    directories: Vec<PathBuf>,
+}
+
+fn is_whisper_file(path: &Path) -> bool {
+    path.extension() == Some(std::ffi::OsStr::new("wsp"))
+}
+
+fn collect_whisper_files(dir: &Path) -> Vec<PathBuf> {
+    WalkDir::new(dir)
+        .min_depth(1)
+        .into_iter()
+        .filter_map(|entry| match entry {
+            Ok(ref entry) if is_whisper_file(entry.path()) => Some(entry.path().to_path_buf()),
+            _ => None
+        })
+        .collect::<Vec<PathBuf>>()
+}
+
+fn walk_dir(dir: &Path, delete_corrupt: bool, verbose: bool) {
+    if verbose {
+        println!("Scanning {:?}...", dir);
+    }
+
+    let whisper_files = collect_whisper_files(dir);
+    for file in &whisper_files {
+        match whisper::WhisperFile::open(file) {
+            Ok(whisper_file) => {
+                let x: u32 = whisper_file.info().archives.iter().map(|a| a.points).sum();
+                println!("{:?}: {} points", file, x);
+            }
+            _ => {
```

Comment:
friendly advice to catch and print error.

Reply:
this is done similar to python version

Reply:
Looking closer here i think we do not  need to catch and print, because all errors in this case leads us to corrupt file.

Reply:
Not really. There is a possible issue with io race-condition, when the directory is absent(deleted) and trying to delete the file in this case is incorrect. This is a known behaviour in find utility.